### PR TITLE
Redesign infrastructure and endpoints. Expand ProtectionService capabilities.

### PR DESCRIPTION
**Pull request by micmarty**
- Redesign infrastructure -> use UUID as keys. Stop relying on GPU index (which can vary after reboots), because UUID is fixed and globally unique .
- Provide additional info about metric's unit  -> e.g. ```'fan_speed': {'value': 30, 'unit': '%'}```
- Add 'index' parameter to ```nvidia-smi --query-gpu```
- Monitor active GPU processes on each node and send a warning message **only to those users who violate someone else's reservation by running own GPU process**.
- Ignore some system processes, like: Xorg, X etc.
- Read reservation events from DB, instead of mocked object.
- Move 'name' and 'index' parameters outside 'metrics'
- Redesign ```/metrics``` endpoint by splitting it into ```/info```, ```/metrics```, ```/processes```
- Handle some edge cases + log to stdout

ProtectionService was testes on **des** host.

1. ```tensorhive db init```
2.  Run tensorhive on info level -> ```tensorhive --log-level debug```
3. Open Swagger UI in web browser. Create some fake user (which has different username than yours), then create reservation event for that user. As ```resourceId``` put UUID of GPU you want to test on. Remember to select valid **start** and **end**
4. Run some process on that GPU via ssh
5. You should get warning messages in all of your remote shell sessions, because you have violated reservation of fake user
6. After **end** or after your process terminates, messages should appear no more

### CODE REVIEW

**Comment by micmarty** [comment]
Closing issues: #53 #57 #58

File: tensorhive/core/services/ProtectionService.py [hunk 1]
```diff
@@ -70,39 +103,65 @@ def as_dict(line):
...
             # 2. Establish connection to node and find all tty sessions
             node_connection = self.connection_manager.single_connection(hostname)
             node_sessions = self.node_tty_sessions(node_connection)
+            node_processes = self.node_gpu_processes(hostname)
+            process_owners = self.process_owners_on_node(node_processes)
 
             # 3. Any session that does not belong to a priviliged user should be rembered
             for session in node_sessions:
-                if session['USER'] != username:
+                if (session['USER'] != username) and (session['USER'] in process_owners):
```
Comment (roscisz) [inline tensorhive/core/services/ProtectionService.py]: Is the specific id for the reserved device matched with the id of the device used by the process?

File: tensorhive/core/services/ProtectionService.py [hunk 2]
```diff
@@ -70,39 +103,58 @@ def as_dict(line):
...
+                log.warning('Unable to process the reservation ({}@{}), skipping...'.format(username, hostname))
+                continue
 
             # 2. Establish connection to node and find all tty sessions
             node_connection = self.connection_manager.single_connection(hostname)
             node_sessions = self.node_tty_sessions(node_connection)
+            node_processes = self.node_gpu_processes(hostname)
+            process_owners = self.process_owners_on_node(node_processes)
 
             # 3. Any session that does not belong to a priviliged user should be rembered
```
[the remark this answers is not in the document]
Comment (micmarty) [reply tensorhive/core/services/ProtectionService.py]: Done.

File: tensorhive/core/services/ProtectionService.py (hunk 1 above)
Comment (micmarty) [reply tensorhive/core/services/ProtectionService.py]: Fixed, now it should be working as expected.

**Pull request merged**